codeql: 2.8.2 -> 2.8.5

anticomputer opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

- update: v2.8.2 to v2.8.5
- bugfix: support codeql database creation on nixos linux64 by swapping out the binary packaged java distribution for the nixos jdk11 distribution

The existing nixpkgs codeql package performs a sed operation to set a static path to nixpkgs jdk11 for the main codeql executable. However there are many other executables that need to be able to resolve a functioning java home inside the codeql binary distribution, including database extractors and autobuilders and they also depend on CODEQL_DIST and CODEQL_PLATFORM to resolve java home.

This update swaps out the codeql java distribution in its entirety by symlinking against nixpkgs jdk11 as well as enabling the autopatchElfhook such that all database extractors and autobuilders actually function on nixos. This means CODEQL_DIST and  CODEQL_PLATFORM are consistently correct for all binary dependencies inside the codeql distribution.

Prior to this update you could only use codeql main commands, i.e. run queries. This version of the package also enables you to create new databases.

Known issues: 

The csharp extractor depends on an old version of libttng-ust (.so.0), nixpkgs currently ships a more modern version (.so.1), which makes autopatchElf fail against the csharp extractor. This means csharp extraction currently does not work on nixpkgs codeql.  This update ignores this dependency failure. All other extractors and autobuilders patch and function correctly. Once the codeql csharp extractor updates to a more recent libttng-ust dependency in a future release, this will automatically resolve itself.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

anticomputer committed:
codeql: 2.8.2 -> 2.8.5

- update: v2.8.2 to v2.8.5
- bugfix: support codeql database creation on nixos linux64

anticomputer committed:
remove trailing whitespace

drewrisinger commented on pkgs/development/tools/analysis/codeql/default.nix:
```diff
@@ -34,10 +38,17 @@ stdenv.mkDerivation rec {
 
     ln -s $out/codeql/codeql $out/bin/
-  '';
+    '';
```
```suggestion
  '';
```

drewrisinger commented on pkgs/development/tools/analysis/codeql/default.nix:
```diff
@@ -2,17 +2,20 @@
   };
 
+  # needed until codeql/csharp/tools/linux64/libcoreclrtraceptprovider.so updates its liblttng-ust dependency from liblttng-ust.so.0 to liblttng-ust.so.1
+  autoPatchelfIgnoreMissingDeps=true;
```
```suggestion
  autoPatchelfIgnoreMissingDeps = true;
```

drewrisinger reviewed (approved):
* [x] Diff LGTM
* [x] Commit LGTM
* [x] Builds via ``nix-review`` on x86-64-linux

```
1 package built:
codeql
```

anticomputer committed:
Update pkgs/development/tools/analysis/codeql/default.nix

Co-authored-by: Drew <drewrisinger@users.noreply.github.com>

anticomputer committed:
Update pkgs/development/tools/analysis/codeql/default.nix

Co-authored-by: Drew <drewrisinger@users.noreply.github.com>

anticomputer committed:
add lttng-ust as native build input

add lttng-ust as native build input such that the csharp dependency issue will automatically resolve when updated by codeql maintainers

drewrisinger commented on pkgs/development/tools/analysis/codeql/default.nix:
```diff
@@ -27,6 +27,7 @@ stdenv.mkDerivation rec {
     jdk11
     stdenv.cc.cc.lib
     curl
+    lttng-ust
```
I'm not sure how I feel about adding a speculative dependency when it's not strictly necessary for today's build. Increases the closure for no particular reason. Seems more prudent to just add it when the upstream maintainers change their build process. Also, is there a reference upstream issue/patch to link to?

anticomputer replied:
Fair callout. I'm in convo with the maintainers, will share back a public issue when it is available, will remove the extra dependency for now.

anticomputer committed:
remove lttng-ust dependency

as per review, remove lttng-ust dependency until library dependency issues are resolved upstream, re-add in a future update

anticomputer commented on pkgs/development/tools/analysis/codeql/default.nix:
```diff
@@ -34,7 +38,14 @@ stdenv.mkDerivation rec {
+    # the codeql packaged java dist, but we DO want to patch the extractors
+    # as well as the builders which are ELF binaries for the most part
+
+    rm -rf $out/codeql/tools/linux64/java
```
@SuperSandro2000 it's a directory

SuperSandro2000 commented:
please rebase

wegank commented:
Closed by #168228.
